packages/react-native-reanimated/src/initializers.ts

```diff
@@ -1,18 +1,50 @@
 'use strict';
-import { reportFatalErrorOnJS } from './errors';
+import { registerReanimatedError, reportFatalErrorOnJS } from './errors';
 import { isChromeDebugger, isJest, shouldBeUseWeb } from './PlatformChecker';
 import {
   runOnJS,
   setupMicrotasks,
   callMicrotasks,
   runOnUIImmediately,
+  executeOnUIRuntimeSync,
 } from './threads';
 import { mockedRequestAnimationFrame } from './mockedRequestAnimationFrame';
+import { logger, logToLogBoxAndConsole, updateLoggerConfig } from './logger';
+import { makeShareableCloneRecursive } from './shareables';
+import { shareableMappingCache } from './shareableMappingCache';
 
 const IS_JEST = isJest();
 const SHOULD_BE_USE_WEB = shouldBeUseWeb();
 const IS_CHROME_DEBUGGER = isChromeDebugger();
 
+// this is for web implementation
+if (SHOULD_BE_USE_WEB) {
+  global._WORKLET = false;
+  global._log = console.log;
+  global._getAnimationTimestamp = () => performance.now();
+}
+
+// Register ReanimatedError and update logger config in the UI global scope.
+// (we are using `executeOnUIRuntimeSync` here to make sure that the changes
+// are applied before any async operations are executed on the UI runtime)
+if (!shouldBeUseWeb()) {
+  // Register ReanimatedError in the UI global scope
+  executeOnUIRuntimeSync(registerReanimatedError)();
+  // Register logger config in the UI global scope
+  executeOnUIRuntimeSync(updateLoggerConfig)();
+}
+
+// Override the logFunction implementation with the one that adds logs
+// with better stack traces to the LogBox (need to override it after `runOnJS`
+// is defined).
+updateLoggerConfig({
+  logFunction(data) {
+    'worklet';
+    runOnJS(logToLogBoxAndConsole)(data);
+  },
+});
+shareableMappingCache.set(logger, makeShareableCloneRecursive(logger));
```

Comment:
I don't understand. You call `updateLoggerConfig` via `executeOnUIRuntimeSync` and then you call it again without it. Could you explain?

Reply:
### The first call

```tsx
executeOnUIRuntimeSync(updateLoggerConfig)();
```

Is needed to create the logger config object on the UI runtime. What `updateLoggerConfig` does without any parameter when called for the first time, before the `__reanimatedLoggerConfig` exists in the UI runtime, is creating the config object and registering it in the global scope:

```tsx
export function updateLoggerConfig(options?: Partial<LoggerConfigInternal>) {
  'worklet';
  const config = _WORKLET
    ? ((global as Record<string, unknown>)
        .__reanimatedLoggerConfig as LoggerConfigInternal) ?? {} // <-- this empty object will be returned
    : __reanimatedLoggerConfig;

  // <-- will be filled with default values from below
  config.logFunction =
    config?.logFunction ?? options?.logFunction ?? logToConsole; // <-- `logToConsole`
  config.level = options?.level ?? LogLevel.warn; // <-- `LogLevel.warn`
  config.strict = options?.strict ?? false; // <-- `false`

  if (_WORKLET) { // <-- Because it is called with `executeOnUIRuntimeSync`, it is a worklet
    registerLoggerConfig(config); // <-- and will store the config object in the UI runtime global scope
  }
}
```

---

### The second call

It is executed on the React runtime and it replaces the `logFunction` in the config.

```tsx
updateLoggerConfig({
  logFunction(data) {
    'worklet';
    runOnJS(logToLogBoxAndConsole)(data);
  },
});
```

Reply:
But, I can see now that this is invalid as `logFunction` is always the default implementation then instead of the `logToLogBoxAndConsole`